src/version/features/ContrastMode/index.js

```diff
@@ -0,0 +1,57 @@
+import React, { Component } from "react";
+import { connect } from 'react-redux';
+
+import IconButton from '@material-ui/core/IconButton';
+import ContrastIcon from '@material-ui/icons/Tonality';
+
+import { contrastModeAction } from "../../actions/contrastModeAction";
+
+/**
+ * Thic component returns Contrast Mode button
+ *
+ * @author Bogdan Shcherban <bsc@piogroup.net>
+ */
+class ContrastMode extends Component {
+
+    state = {
+        isContrastMode: this.props.contrastMode,
+    };
+
+    toggleContrastMode = () => {
+        this.setState(
+            { isContrastMode: !this.state.isContrastMode },
```

Comment:
again, when you update the state depending on the previous state, use the function form or you risk race conditions

```js
this.setState(state => ({ isContrastMode: !!state.isContrastMode }), ...)
```

Reply:
Fixed